### PR TITLE
Documentation - Add more tips to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by Paliak**
- Mention compiling emmyLua form source
- Mention excluding directories from emmyLua language server
- Mention misc tips
- Mention running with wine
- Mention forcing Dev mode off to debug upgrade system
- Mention that emmyLua requires java

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -148,6 +150,26 @@ dbg.tcpListen("localhost", 9966)
 5. Start Path of Building Community
 6. Attach the debugger
 
+#### Excluding directories from emmyLua
+
+Depending on the amount of system ram you have avilable and the amout that gets assinged to the jvm running the emmylua language server you might run into issues when trying to debug Path of building.
```
Comment (Nightblade) [inline CONTRIBUTING.md:155]: Depending on the amount of system ram you have available and the amount that gets assigned to the jvm running the emmylua language server you might run into issues when trying to debug Path of building.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -157,6 +179,12 @@ dbg.tcpListen("localhost", 9966)
 5. Start Path of Building Community
 6. Attach the debugger
 
+#### Miscellaneous tips
+If you're on windows, consider downloading [git for windows](https://git-scm.com/downloads) and installing git bash. Git bash comes with a variety of typical linux tools such as grep that can make navigating the code base much easier.
+
+If you're using linux you can run the `./runtime/Path{space}of{space}Building.exe` executable with wine. You will need to provide a valid wine path to the emmy lua debuger directory.
```
Comment (Nightblade) [inline CONTRIBUTING.md:185]: If you're using linux you can run the `./runtime/Path{space}of{space}Building.exe` executable with wine. You will need to provide a valid wine path to the emmy lua debugger directory.

(spelling)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -130,6 +130,8 @@ It is recommended to use it over the built-in Lua plugins.
 Please note that EmmyLua is not available for other editors based on Visual Studio Code,
 such as [VSCodium](https://vscodium.com) or [Eclipse Theia](https://theia-ide.org) but can be built from source if needed.
 
+Note that you will need to have java installed to use emmyLua, and have the JAVA_HOME environment variable correctly setup.
```
Comment (Nightblade) [inline CONTRIBUTING.md:133]: The environment variable wouldn't work for me in VSCode, so I ended up just adding this to my `settings.json` :
```json
"emmylua.java.home": "C:/Program Files (x86)/Java/jre1.8.0_201/",
```

**Comment by Paliak** [comment]
Thanks for proof reading, Should be fixed now.

**Review by Wires77 — APPROVED** [review]
Thanks for updating documentation!

**Pull request merged**